letsencrypt/client/network2.py

```diff
@@ -280,26 +321,19 @@ def answer_challenge(self, challb, response):
         try:
             authzr_uri = response.links['up']['url']
         except KeyError:
-            raise errors.NetworkError('"up" Link header missing')
+            # TODO: Right now Boulder responds with the authorization resource
+            # instead of a challenge resource... this can be uncommented
+            # once the error is fixed.
+            return None
+            # raise errors.NetworkError('"up" Link header missing')
         challr = messages2.ChallengeResource(
             authzr_uri=authzr_uri,
             body=messages2.ChallengeBody.from_json(response.json()))
         # TODO: check that challr.uri == response.headers['Location']?
         if challr.uri != challb.uri:
-            raise errors.UnexpectedUpdate(challr.uri)
+            raise errors.UnexpectedUpdate(challb.uri)
```

Comment:
Sorry... agreed... assuming the server will respond with a challenge resource.  I believe the spec contradicts itself on this... it is 

```The server MUST ignore any fields in the response object that are not specified as response fields for this type of challenge. The server provides a 200 response including the updated challenge.```

Early in the spec it specifically mentions the challenge resource and uplink... immediately after the responding to challenges section... it sounds like an authorization resource is returned.  What we really should have is an available GET request for the challenge resource (as the validated status won't be available immediately).  Will revert it back... (this PR was a major pain with all of the outstanding boulder/spec issues...)